tests/src/whisk/core/cli/test/WskCoreBasicTests.scala

```diff
@@ -0,0 +1,287 @@
+/*
+ * Copyright 2015-2016 IBM Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package whisk.core.cli.test
+
+import org.junit.runner.RunWith
+import org.scalatest.junit.JUnitRunner
+
+import common.RunWskAdminCmd
+import common.TestHelpers
+import common.TestUtils
+import common.TestUtils.FORBIDDEN
+import common.TestUtils.NOT_FOUND
+import common.TestUtils.TIMEOUT
+import common.Wsk
+import common.WskProps
+import common.WskTestHelpers
+import spray.json.DefaultJsonProtocol._
+import spray.json.pimpAny
+import whisk.core.entity.Subject
+import whisk.core.entity.WhiskPackage
+
+@RunWith(classOf[JUnitRunner])
+class WskCoreBasicTests
+    extends TestHelpers
+    with WskTestHelpers {
+
+    implicit val wskprops = WskProps()
+    val wsk = new Wsk(usePythonCLI = false)
+    val samplePackage = "samplePackage"
+    val sampleAction = samplePackage + "/sampleAction"
+    val sampleFeed = samplePackage + "/sampleFeed"
+    val wskadmin = new RunWskAdminCmd {}
+
```

Comment:
@rabbah  The sample you offered does not work, because the test suite only recognizes implicit WskProps, and otherWskProps defined here does not take effect.
We have to use only one WskProps, and reassign it to the value with different auth keys, as the old way I did in this patch.

If we want to reuse some common code, I actually do not see their advantages over the way I override beforeall to create all the package, action and feed at the beginning and remove all of them by overriding afterall in the end.